src/main/java/seedu/address/model/loan/LoanTime.java

```diff
@@ -35,133 +32,202 @@
+     * @return true if the string is a valid Time
      */
     public static boolean isValidTime(String test) {
         String[] timeData = test.split(":");
-        if (timeData.length <= 1) {
-            return false;
-        }
+        assert timeData.length == 2;
+
         return (Integer.parseInt(timeData[0]) < 24) && (Integer.parseInt(timeData[1]) < 60);
     }
 
     /**
      * Returns the difference in time given one other LoanTime object.
-     * This will be returned as a long of number of second.
+     * This will be returned as a long of number of minutes.
      * Function returns 0 if the specified other LoanTime is before the current LoanTime.
      *
      * @param otherTime LoanTime object to be compared to.
      */
     public long loanTimeDifferenceMinutes(LoanTime otherTime) {
-        long timeDifference = (otherTime.value.toEpochMilli() - this.value.toEpochMilli());
-        return (timeDifference >= 0) ? timeDifference / 60000 : 0;
+        return loanTimeDifferenceMinutes(this, otherTime);
     }
 
     /**
      * Returns the difference in time given two LoanTime objects.
-     * This will be returned as a long of number of second.
+     * This will be returned as a long of number of minutes.
      * Function returns 0 if the specified other LoanTime is before the current LoanTime.
      *
      * @param currentTime LoanTime object to signify start of time interval.
      * @param otherTime   LoanTime object to signify end of time interval.
      */
     public static long loanTimeDifferenceMinutes(LoanTime currentTime, LoanTime otherTime) {
-        long timeDifference = (otherTime.value.toEpochMilli() - currentTime.value.toEpochMilli());
-        return (timeDifference >= 0) ? timeDifference / 60000 : 0;
+        long timeDifference = Duration.between(currentTime.value, otherTime.value).toMinutes();
+        return (timeDifference >= 0) ? timeDifference : 0;
+    }
+
+    /**
+     * Parses the specified string into an {@link Instant}. This method will use the current System
+     * time zone during the parsing process.
+     *
+     * @param objString The string to parse
+     * @return The parsed {@link Instant}
+     */
+    private static Instant parseLoanTime(String objString) {
+        // If the string is not valid, throw an exception.
+        if (!isValidLoanTime(objString)) {
+            throw new IllegalArgumentException(MESSAGE_LOANTIME_CONSTRAINTS);
+        }
+
+        // Prepare the Date and Time fields.
+        LocalDate loanLocalDate;
+        LocalTime loanLocalTime;
+
+        // Consider both cases.
+        if (isValidLongLoanTimeFormat(objString)) {
+            // Split by spaces to separate the Date and Time strings
+            String[] dateData = objString.split(" ");
+
+            assert dateData.length == 2;
+
+            // Obtain the Date and Time strings
+            String loanDateString = dateData[0];
+            String loanTimeString = dateData[1];
```

Comment:
I noticed that this is the same process as line 83. Perhaps some abstraction/method extraction is appropriate.